sdk/cosmos/azure-spring-data-cosmos-core/src/main/java/com/azure/spring/data/cosmos/repository/query/ReactiveCosmosQueryMethod.java

```diff
@@ -56,4 +61,33 @@ public ReactiveCosmosQueryMethod(Method method, RepositoryMetadata metadata, Pro
     private static boolean isReactiveWrapperClass(Class<?> clazz) {
         return clazz.equals(Flux.class) || clazz.equals(Mono.class);
     }
+
+    /**
+     * Returns whether the method has an annotated query.
+     * @return if the query method has an annotated query
+     */
+    public boolean hasAnnotatedQuery() {
+        return findAnnotatedQuery().isPresent();
+    }
+
+    /**
+     * Gets the annotated query or returns null
+     * @return the annotated query String or null
+     */
+    @Nullable
+    public String getQueryAnnotation() {
+        return findAnnotatedQuery().orElse(null);
+    }
+
+    private Optional<String> findAnnotatedQuery() {
+
+        return lookupQueryAnnotation()
+                   .map(Query::value)
+                   .filter(StringUtils::hasText);
```

Comment:
Same here, we can use a field `stringInQueryAnnotation ` instead of calculate it  every time.

Reply:
Refactored